packages/@aws-cdk/aws-servicecatalog/lib/product.ts

```diff
@@ -202,20 +214,46 @@ export class CloudFormationProduct extends Product {
     }
   }
 
-  private renderProvisioningArtifacts(
-    props: CloudFormationProductProps): CfnCloudFormationProduct.ProvisioningArtifactPropertiesProperty[] {
-    return props.productVersions.map(productVersion => {
+  private renderProvisioningArtifacts(props: CloudFormationProductProps): CfnCloudFormationProduct.ProvisioningArtifactPropertiesProperty[] {
+    let productVersions: CfnCloudFormationProduct.ProvisioningArtifactPropertiesProperty[] = [];
+    for (const productVersion of props.productVersions) {
       const template = productVersion.cloudFormationTemplate.bind(this);
-      InputValidator.validateUrl(this.node.path, 'provisioning template url', template.httpUrl);
-      return {
-        name: productVersion.productVersionName,
-        description: productVersion.description,
-        disableTemplateValidation: productVersion.validateTemplate === false ? true : false,
-        info: {
-          LoadTemplateFromURL: template.httpUrl,
+      let httpUrl = template.httpUrl;
+      switch (template.templateType) {
+        case TemplateType.PRODUCT_STACK:
+          if (template.productVersionDetails) {
+            template.productVersionDetails.productPathUniqueId = this.productPathUniqueId;
+            template.productVersionDetails.productVersionName = productVersion.productVersionName;
+            template.productVersionDetails.versioningStrategy = productVersion.versioningStrategy;
+          }
+          break;
+        case TemplateType.PRODUCT_STACK_CONTEXT:
+          const templateFileKey = `${this.productPathUniqueId}.${template.productVersionDetails?.productStackId}.${productVersion.productVersionName}.product.template.json`;
+          const templateFilePath = path.join(PRODUCT_STACK_CONTEXT_DIRECTORY, templateFileKey);
+          if (!fs.existsSync(templateFilePath)) {
+            throw new Error(`Template ${templateFileKey} cannot be found in product stack context`);
+          }
+          httpUrl = new s3_assets.Asset(this, `Template${hashValues(templateFileKey)}`, {
+            path: templateFilePath,
+          }).httpUrl;
+          break;
+        default:
+          break;
+      }
+
+      InputValidator.validateUrl(this.node.path, 'provisioning template url', httpUrl);
+      productVersions.push(
+        {
+          name: productVersion.productVersionName,
+          description: productVersion.description,
+          disableTemplateValidation: productVersion.validateTemplate === false ? true : false,
+          info: {
+            LoadTemplateFromURL: httpUrl,
+          },
         },
-      };
-    });
+      );
+    }
```

Comment:
```suggestion
      });
```

Reply:
ESLint is not happy and suggesting the original braces here. 
Also the suggestion is moving the bracket before the parenthesis which doesn't make sense.

Reply:
ahh woops that was the wrong pair